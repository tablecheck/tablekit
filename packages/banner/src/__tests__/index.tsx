import { mount } from '@tablecheck/tablekit-enzyme';
import '@tablecheck/tablekit-free-icon-config';

import { Banner } from '../index';

beforeEach(() => {
  global.console.error = jest.fn();
});

describe('Banner', () => {
  test('should be possible to create a component', () => {
    const wrapper = mount(<Banner message="Test" />);
    expect(wrapper).not.toBeUndefined();
  });

  test('should correctly apply styling', () => {
    expect(mount(<Banner message="Test" />)).toMatchSnapshot();
  });

  test('should allow passing of custom className', () => {
    expect(
      mount(<Banner className="test-class" message="Test" />)
    ).toMatchSnapshot();
  });
});
