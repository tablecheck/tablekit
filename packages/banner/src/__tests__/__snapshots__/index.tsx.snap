// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should allow passing of custom className 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 4px 0 16px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  font-size: 16px;
  line-height: 24px;
  font-weight: 600;
  background-color: #7935D2;
  color: white;
}

.emotion-0 .bannerIcon path {
  fill: white;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 6px 0;
}

.emotion-2:only-child {
  padding: 6px 8px 6px 0;
}

.emotion-4 {
  margin-right: 8px;
}

<Banner
  className="test-class"
  message="Test"
>
  <BannerContainer
    appearance="primary"
    className="test-class"
    size="regular"
  >
    <Noop />
    <div
      className="test-class emotion-0 emotion-1"
      size="regular"
    >
      <BannerMessageContainer>
        <Noop />
        <div
          className="emotion-2 emotion-3"
        >
          <AlertIcon
            className="bannerIcon"
            icon="fas-info-circle"
            size="regular2"
          >
            <Noop />
            <Icon
              className="bannerIcon emotion-4 emotion-5"
              icon="fas-info-circle"
              size="regular2"
            >
              icon-info-circle
            </Icon>
          </AlertIcon>
          <div>
            Test
          </div>
        </div>
      </BannerMessageContainer>
    </div>
  </BannerContainer>
</Banner>
`;

exports[`Banner should correctly apply styling 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 4px 0 16px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  font-size: 16px;
  line-height: 24px;
  font-weight: 600;
  background-color: #7935D2;
  color: white;
}

.emotion-0 .bannerIcon path {
  fill: white;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  padding: 6px 0;
}

.emotion-2:only-child {
  padding: 6px 8px 6px 0;
}

.emotion-4 {
  margin-right: 8px;
}

<Banner
  message="Test"
>
  <BannerContainer
    appearance="primary"
    size="regular"
  >
    <Noop />
    <div
      className="emotion-0 emotion-1"
      size="regular"
    >
      <BannerMessageContainer>
        <Noop />
        <div
          className="emotion-2 emotion-3"
        >
          <AlertIcon
            className="bannerIcon"
            icon="fas-info-circle"
            size="regular2"
          >
            <Noop />
            <Icon
              className="bannerIcon emotion-4 emotion-5"
              icon="fas-info-circle"
              size="regular2"
            >
              icon-info-circle
            </Icon>
          </AlertIcon>
          <div>
            Test
          </div>
        </div>
      </BannerMessageContainer>
    </div>
  </BannerContainer>
</Banner>
`;
